transport/src/main/java/com/strategyobject/substrateclient/transport/ws/ReconnectionPolicy.java

```diff
@@ -0,0 +1,43 @@
+package com.strategyobject.substrateclient.transport.ws;
+
+import lombok.NonNull;
+
+/**
+ * Represents a strategy of reconnection
+ */
+public interface ReconnectionPolicy {
+
+    /**
+     * The method is called when connection was closed and probably should be reconnected
+     * @param context contains a reason of disconnection
+     * @return a schedule planned for reconnection
+     */
+    @NonNull ReconnectionSchedule apply(@NonNull ReconnectionContext context);
```

Comment:
I don't see the context being used anywhere. Also probably the policy should be affected by the reason of disconnection.

Reply:
Sorry, I didn't get your point.